### PR TITLE
Forward all throwables but not only excpetions - Fix the CI timeout

### PR DESCRIPTION
**Pull request by beeender**
For example the UnsatisfiedLinkError should cause the process terminated.
This is the reason of all the time out issues we found for CI.

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/HandlerController.java [hunk 1]
```diff
@@ -44,13 +44,15 @@
 /**
  * Centralises all Handler callbacks, including updating async queries and refreshing the Realm.
  */
-final class HandlerController implements Handler.Callback {
+public final class HandlerController implements Handler.Callback {
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/HandlerController.java]: Why public? This will make it part of our public API 😢

**Comment by cmelchior** [comment]
Great catch 💯 , but we should really avoid making HandlerController public.

File: realm/realm-library/src/main/java/io/realm/HandlerController.java (hunk 1 above)
Comment (beeender) [inline realm/realm-library/src/main/java/io/realm/HandlerController.java]: uh, yes, forgot about the API ... let me find a better a way to do it.

**Comment by zaki50** [comment]
nice!

File: realm/realm-library/src/main/java/io/realm/internal/async/QueryUpdateTask.java [hunk 2]
```diff
@@ -100,8 +101,17 @@ public void run() {
                 handler.obtainMessage(message, result).sendToTarget();
             }
 
-        } catch (Exception e) {
+        } catch (BadVersionException e) {
+            // In some rare race conditions, this can happen. In that case, just ignore the error.
```
Comment (beeender) [inline realm/realm-library/src/main/java/io/realm/internal/async/QueryUpdateTask.java]: @cmelchior @nhachicha I believe we should cat it here as well, right?

**Comment by zaki50** [comment]
:+1: Great!

File: realm/realm-library/src/main/java/io/realm/internal/HandlerControllerConstants.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal;
+
+public final class HandlerControllerConstants {
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/internal/HandlerControllerConstants.java]: Add a small Javadoc stating the purpose of this class.

File: realm/realm-library/src/main/java/io/realm/internal/async/QueryUpdateTask.java (hunk 2 above)
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/internal/async/QueryUpdateTask.java]: Good catch. Yes it should. Although with 1.2, the handover object should now prevent the version from being invalidated, so I think we can actually remove all our code that handles the BadVersionException, but I would do that in another PR as it needs testing.

**Comment by cmelchior** [comment]
Apart from missing Javadoc 👍
